dbt/adapters/athena/impl.py

```diff
@@ -58,14 +58,19 @@ def clean_up_partitions(
         with boto3_client_lock:
             glue_client = boto3.client('glue', region_name=client.region_name)
         s3_resource = boto3.resource('s3', region_name=client.region_name)
-        partitions = glue_client.get_partitions(
-            # CatalogId='123456789012', # Need to make this configurable if it is different from default AWS Account ID
-            DatabaseName=database_name,
-            TableName=table_name,
-            Expression=where_condition
-        )
-        p = re.compile('s3://([^/]*)/(.*)')
-        for partition in partitions["Partitions"]:
+        paginator = glue_client.get_paginator("get_partitions")
```

Comment:
if you have a chance, can you check what would happen if the current user doesn't have enough permission to fetch the partitions?

Reply:
You bet. I should have some time to look into it tomorrow to check the permission case and run through the proper dbt-athena testing.

Reply:
@DVerzal
Thank you.
While I debug #84, I noticed that there could be unexpected behaviors when Error in API is not handled properly :D

I am working on making a PR to handle #84

Reply:
@DVerzal this is an issue due to missing permissions. You are missing a `glue:GetPartitions` on the table.